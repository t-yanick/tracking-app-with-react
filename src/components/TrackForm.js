import React, { useState } from 'react';
import PropTypes from 'prop-types';
// import pluralize from 'pluralize';

const TrackForm = ({
  items, targetDate,
}) => {
  const [error, setError] = useState('');
  const [date, setDate] = useState('');
  const [state, setState] = useState('');

  const onDateChange = (e) => {
    setDate(e.target.value);
  }

  const handleInputChange = () => {

  };

  const onSubmit = (e) => {
    e.preventDefault();
  };

  return (
    <div className="track-form mb3">
      display error message here
      <form className="">
        <div className="">
          <input type="date" onChange={onDateChange} value={date} />
        </div>
        <div className="track-form__group mb3">
          {/* {items.map((item) => (
            <div className="track-form__item" key={item.id}>
              <div className="track-form__icon">
                <span className="iconify" data-icon={item.icon} data-inline="false" />
              </div>
              <div className="track-form__title">{item.title}</div>
              <input
                type="number"
                name={item.title}
                className="track-form__input"
                maxLength="4"
                onChange={(e) => handleInputChange(e.target.value, item.id)}
                value={state[item.id]}
              />
              <div className="track-form__unit">plural of units</div>
              <div className="track-form__target">target of item</div>
            </div>
          )) } */}
        </div>
        <button type="submit" className="btn dark">Save</button>
      </form>
    </div>
  );
};

export default TrackForm;