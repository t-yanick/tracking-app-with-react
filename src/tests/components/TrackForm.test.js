import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TrackForm from '../../components/TrackForm';
import createTestStore from '../fixtures/createTestStore';

describe('TrackForm', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <TrackForm />
          </MemoryRouter>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});