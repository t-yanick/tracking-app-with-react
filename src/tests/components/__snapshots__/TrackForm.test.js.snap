// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackForm should match with snapshot 1`] = `
<div
  className="track-form mb3"
>
  
  <form
    className="track-form__form mb3"
    onSubmit={[Function]}
  >
    <div
      className="track-form__date"
    >
      <input
        onChange={[Function]}
        type="date"
        value="2021-10-04"
      />
    </div>
    <div
      className="track-form__group mb3"
    />
    <button
      className="btn dark"
      type="submit"
    >
      Save
    </button>
  </form>
</div>
`;
